src/app/global/utils.nim

```diff
@@ -142,5 +142,8 @@ QtObject:
   proc getColorHashAsJson*(self: Utils, publicKey: string): string {.slot.} =
     procs_from_visual_identity_service.getColorHashAsJson(publicKey)
 
+  proc getColorId*(self: Utils, publicKey: string): int {.slot.} =
+    int(procs_from_visual_identity_service.colorIdOf(publicKey))
```

Comment:
Do those two functions request data from the DB? If so, it should probably be handled in Nim somewhere with a cache, since hitting the DB every time we need hash or color ID is not very efficient. 

The color ID should be carried in the required model, for example a contact model or the one for the chat.

If you really need to get the ID on the fly, it might be better to store them in a cache in the visual_identity_service instead of requesting it from the DB.

Reply:
Good concerns but these are not database queries. These are just proxies for stateless status-go methods (take pubkey return colorHash or colorId).

There is only one scenario in which the colorHash and colorId are populated from DB, this is when we are not able to obtain pubkey. The login page is using unencrypted accounts data to display users you select from, this data does not (and should not) contain pubkey information, it is only accessible after login.

```diff
index 78b6ff05..f8fa7d37 100644
--- a/multiaccounts/database.go
+++ b/multiaccounts/database.go
@@ -3,6 +3,7 @@ package multiaccounts
 import (
     "context"
     "database/sql"
+    "encoding/json"
 
     "github.com/ethereum/go-ethereum/log"
     "github.com/status-im/status-go/images"
@@ -15,6 +16,8 @@ type Account struct {
     Name           string                 `json:"name"`
     Timestamp      int64                  `json:"timestamp"`
     Identicon      string                 `json:"identicon"`
+    ColorHash      [][]int                `json:"colorHash"`
+    ColorId        int64                  `json:"colorId"`
     KeycardPairing string                 `json:"keycard-pairing"`
     KeyUID         string                 `json:"key-uid"`
     Images         []images.IdentityImage `json:"images"`
```

Reply:
These could be thought of as utils because they are stateless and deterministic. In other words, the result of `colorId = colorIdOf(publicKey)` will be always the same for the same pubkeys. This is similar to `getIdenticon(publicKey)` we have introduced in the past.

If that is a costly operation we should determine by profiling our app. If that's the case we could employ some caching techniques either on status-go or nim side.